common/utilities/string/wrap-text.h

```diff
@@ -0,0 +1,117 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <sstream>
+
+#include "../third-party/imgui/imgui.h"
+
+namespace utilities {
+namespace string {
+
+
+// Split input test to lines
+//     - Input: std:string text
+//     - Output: a vector of (std::string) lines
+// Example:
+//     Input:
+//          This is the first line\nThis is the second line\nThis is the last line
+//     Output:
+//          [0] this is the first line
+//          [1] this is the second line
+//          [2] this is the last line
+inline std::vector< std::string > split_string_by_newline( const std::string & str )
+{
+    auto result = std::vector< std::string >{};
+    auto ss = std::stringstream{ str };
+
+    for( std::string line; std::getline( ss, line, '\n' ); )
+        result.push_back( line );
+
+    return result;
+}
+
+// Wrap text according to input width
+//     - Input:  - text as a string
+//               - wrapping width
+//     - Output: - on success - wrapped text
+//               - on failure - an empty string
+// Example:
+//     Input:
+//          This is the first line\nThis is the second line\nThis is the last line , wrap_width = 15
+//     Output:
+//          this is the\nfirst line\nthis is the\nsecond line\nthis is the\nlast line
+inline std::string wrap_text(const std::string & text, float wrap_pixels_width)
+{
+    // Do not wrap if the wrap is smaller then 32 pixels (~2 characters on font 16)
+    if (wrap_pixels_width < 32.0f)
+        return "";
```

Comment:
An empty string is failure. You say "Do not wrap" so why not return the text, unwrapped?
What is a failure, anyway?